### PR TITLE
fix: make Composer's config.json readable by any user

### LINKED ISSUE #369: /tools/.composer/config.json readable only to root
```
/project # ls -l /tools/.composer/config.json 
-rw-------    1 root     root           155 Jun 20 02:01 /tools/.composer/config.json
```

This makes some Composer plugins (or even Composer itself) unusable.

See https://github.com/sigwinhq/xezilaires-dev/actions/runs/3106723010/jobs/5033958091#step:4:99 for an example.

### PR DESCRIPTION
**Pull request by dkarlovi**
Closes #369.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -186,6 +186,7 @@ RUN --mount=type=secret,id=composer.auth,target=${COMPOSER_HOME}/auth.json \
     --mount=type=secret,id=phive.auth,target=${TOOLBOX_TARGET_DIR}/.phive/auth.xml \
     php toolbox install
 RUN php toolbox test
+RUN chmod 644 /tools/.composer/config.json
```
Comment (dkarlovi) [inline Dockerfile:189]:
```suggestion
RUN chmod 644 ${TOOLBOX_TARGET_DIR}/.composer/config.json
```

maybe?

Comment (jakzal) [reply Dockerfile:189]: Or even:

```dockerfile
RUN chmod 644 ${COMPOSER_HOME}/config.json
```

**Comment by ciaranmcnulty** [comment]
I'm confused, what user are you running it as?

**Comment by dkarlovi** [comment]
@ciaranmcnulty not-root, my host UID.

**Comment by dkarlovi** [comment]
Reported into Composer as https://github.com/composer/composer/issues/11077

**Comment by ciaranmcnulty** [comment]
OK it's deliberate behaviour https://github.com/composer/composer/blob/131da999ac571bcd5bc3dae535dd392d1a36cb53/src/Composer/Config/JsonConfigSource.php#L281

Because we're relocating it and setting COMPOSER_HOME, running as a non-user will try and read it. I guess we should chmod it! Same would apply for auth.json but we don't create one

**Comment by dkarlovi** [comment]
Auth is a secret so meant to be mounted by the user, it shouldn't be handled here IMO. Config is specific since it's configuring the rest of the tooling provided in the image.

**Comment by dkarlovi** [comment]
@ciaranmcnulty how about a bugfix release?

**Comment by ciaranmcnulty** [comment]
Releasing is @jakzal's job

**Comment by ciaranmcnulty** [comment]
Er, although this is actually pushed. Did we maybe not want to do that @jakzal ?

**Comment by jakzal** [comment]
Just triggered the release.

**Comment by dkarlovi** [comment]
Thanks to both! 🍻

**Pull request merged**